e2e_playwright/latest_material_symbol.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import streamlit as st
+
+# Use the material icon from the latest actual material symbol font
+st.success(
+    "Success message to test material icon from latest material symbols font",
+    icon=":material/table_convert:",
+)
+
+st.markdown("Renders correctly in markdown as well: :material/table_convert:")
```

Comment:
nit: not sure if it is worth to add a new e2e script for this, since every additional e2e script has a significant compute overhead compared to putting it into existing e2e tests (and we are already close to the 20 min mark :( ) . I think the `:material/table_convert:` could be just placed somewhere in the `mixed markdown`: 

https://github.com/streamlit/streamlit/blob/2b5f4c0a73383e3e62243e74062deb8ded51f793/e2e_playwright/st_markdown.py#L188

And the success icon could be added somewhere in `st_alert`.  

Also, keeping the snapshots of the same component together in one test makes it a bit easier if we apply visual changes to the component (-> we don't have failures across various tests).

Reply:
We could also add a parameterized python unit test where we add some of the new icon names that is tested against `validate_material_icon`